src/coord/src/catalog.rs

```diff
@@ -2177,6 +2177,13 @@ impl SqlCatalog for ConnCatalog<'_> {
         )
     }
 
+    fn try_get_item_by_id(&self, id: &GlobalId) -> Option<Box<dyn sql::catalog::CatalogItem>> {
+        match self.catalog.try_get_by_id(*id) {
+            Some(item) => Some(Box::new(item.clone())),
+            None => None,
+        }
+    }
```

Comment:
So `Option<&dyn sql::catalog::CatalogItem>`'ll work just fyn here if you explicitly cast to the trait object type!

```patch
diff --git a/src/coord/src/catalog.rs b/src/coord/src/catalog.rs
index c35c96337..1356eca59 100644
--- a/src/coord/src/catalog.rs
+++ b/src/coord/src/catalog.rs
@@ -2177,11 +2177,8 @@ impl SqlCatalog for ConnCatalog<'_> {
         )
     }
 
-    fn try_get_item_by_id(&self, id: &GlobalId) -> Option<Box<dyn sql::catalog::CatalogItem>> {
-        match self.catalog.try_get_by_id(*id) {
-            Some(item) => Some(Box::new(item.clone())),
-            None => None,
-        }
+    fn try_get_item_by_id(&self, id: &GlobalId) -> Option<&dyn sql::catalog::CatalogItem> {
+        self.catalog.try_get_by_id(*id).map(|item| item as &dyn sql::catalog::CatalogItem)
     }
 
     fn get_item_by_id(&self, id: &GlobalId) -> &dyn sql::catalog::CatalogItem {
diff --git a/src/sql/src/catalog.rs b/src/sql/src/catalog.rs
index e82d98648..dab3db6fa 100644
--- a/src/sql/src/catalog.rs
+++ b/src/sql/src/catalog.rs
@@ -109,7 +109,7 @@ pub trait Catalog: fmt::Debug + ExprHumanizer {
     ) -> Box<dyn Iterator<Item = &'a dyn CatalogItem> + 'a>;
 
     /// Gets an item by its ID.
-    fn try_get_item_by_id(&self, id: &GlobalId) -> Option<Box<dyn CatalogItem>>;
+    fn try_get_item_by_id(&self, id: &GlobalId) -> Option<&dyn CatalogItem>;
 
     /// Gets an item by its ID.
     ///
@@ -382,7 +382,7 @@ impl Catalog for DummyCatalog {
         unimplemented!();
     }
 
-    fn try_get_item_by_id(&self, _: &GlobalId) -> Option<Box<dyn CatalogItem>> {
+    fn try_get_item_by_id(&self, _: &GlobalId) -> Option<&dyn CatalogItem> {
         unimplemented!();
     }
 
```